server-app/src/auth/jwt.strategy.ts

```diff
@@ -1,19 +1,80 @@
 import { ExtractJwt, Strategy } from 'passport-jwt';
 import { PassportStrategy } from '@nestjs/passport';
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable } from '@nestjs/common';
 import { jwtConstants } from './constants';
+import { Request } from 'express';
+import { AuthService } from './auth.service';
+import { UsersService } from 'src/users/users.service';
+import { JwtService } from '@nestjs/jwt';
 
 @Injectable()
 export class JwtStrategy extends PassportStrategy(Strategy) {
-  constructor() {
+  constructor(
+    private userService: UsersService,
+    private authService: AuthService,
+    private jwtService: JwtService,
+  ) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
-      ignoreExpiration: false,
-      secretOrKey: jwtConstants.secret,
+      ignoreExpiration: true,
+      passReqToCallback: true,
+      secretOrKey: jwtConstants.SECRET_ACCESS_TOKEN_KEY,
     });
   }
+  async validate(req: Request, payload: any) {
```

Comment:
I would divide this method into smaller ones. You can create a token class or file which would contain all token validatio utilities.